mvpoly/src/utils.rs

```diff
@@ -272,3 +272,35 @@ pub fn compute_indices_nested_loop(nested_loop_sizes: Vec<usize>) -> Vec<Vec<usi
         })
         .collect()
 }
+
+/// Same than `compute_indices_nested_loop` but with an upper bound on the sum
+/// of the indices.
+///
+/// It is useful when the number of nested loop is large and the size of the
+/// loops is large as well. `filter_map` is used to avoid computing the indices
+/// that are not needed and avoid a `capacity_overflow` error.
+pub fn compute_indices_nested_loop_upper_bound(
```

Comment:
I would not expose this function, keep this implementation, and expose compute_indices_nested_loop implemented with that implementation, while computing the upper bound on the fly with a max.
Then we have one function instead of two, which is the efficient variant, and only need one argument